File: ProgramingBasics/SimpleConditionalStatements/08.MetricConverter/src/MetricConverter.java
import java.util.Scanner;

public class MetricConverter {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double number = Double.parseDouble(scanner.nextLine());
        String sourceMetric = scanner.nextLine();
        String destMetric = scanner.nextLine();

        double baseMetricCoefficient = number;
        double coefficient = getCoefficient(sourceMetric);
        baseMetricCoefficient /= coefficient;

        coefficient = getCoefficient(destMetric);
        baseMetricCoefficient *= coefficient;

        System.out.println(baseMetricCoefficient + " " + destMetric);
    }

    private static double getCoefficient(String metric) {
        double result = 0;

        switch (metric) {
            case "m":
                result = 1;
                break;
            case "mm":
                result = 1000;
                break;
            case "cm":
                result = 100;
                break;
            case "mi":
                result = 0.000621371192;
                break;
            case "in":
                result = 39.3700787;
                break;
            case "km":
                result = 0.001;
                break;
            case "ft":
                result = 3.2808399;
                break;
            case "yd":
                result = 1.0936133;
                break;
        }

        return result;
    }
}

File: ProgramingBasics/SimpleConditionalStatements/08.MetricConverter/08.MetricConverter/MetricConverter.cs
using System;

namespace _08.MetricConverter
{
    public class MetricConverter
    {
        public static void Main(string[] args)
        {
            var number = double.Parse(Console.ReadLine());
            var sourceMetric = Console.ReadLine();
            var destMetric = Console.ReadLine();

            var baseMetricCoefficient = number;
            var coefficient = getCoefficient(sourceMetric);
            baseMetricCoefficient /= coefficient;

            coefficient = getCoefficient(destMetric);
            baseMetricCoefficient *= coefficient;

            Console.WriteLine($"{baseMetricCoefficient} {destMetric}");
        }

        private static double getCoefficient(string metric)
        {
            double result = 0;

            if (metric == "m")
            {
                result = 1;
            }
            else if (metric == "mm")
            {
                result = 1000;
            }
            else if (metric == "cm")
            {
                result = 100;
            }
            else if (metric == "mi")
            {
                result = 0.000621371192;
            }
            else if (metric == "in")
            {
                result = 39.3700787;
            }
            else if (metric == "km")
            {
                result = 0.001;
            }
            else if (metric == "ft")
            {
                result = 3.2808399;
            }
            else if (metric == "yd")
            {
                result = 1.0936133;
            }

            return result;
        }
    }
}